pydm/data_plugins/epics_plugins/p4p_plugin_component.py

```diff
@@ -79,9 +81,47 @@ def send_new_value(self, value: Value) -> None:
                 self.write_access_signal.emit(True)
 
             self._value = value
+            has_value_changed_yet = False
             for changed_value in value.changedSet():
-                if changed_value == 'value':
-                    new_value = value.value
+                if changed_value == 'value' or changed_value.split('.')[0] == 'value':
+                    # NTTable has a changedSet item for each column that has changed
+                    # Since we want to send an update on any table change, let's track
+                    # if the value item has been updated yet
+                    if has_value_changed_yet:
+                        continue
+                    else:
+                        has_value_changed_yet = True
+                    
+                    if 'NTTable' in value.getID():
+                        new_value = value.value.todict()
+                    else:
+                        new_value = value.value
+                    
+                    if self.nttable_data_location:
+                        for value in self.nttable_data_location:
+                            if isinstance(new_value, collections.Container) and type(new_value) != str:
+                                if type(value) == str:
+                                    try:
+                                        new_value = new_value[value] 
+                                        continue
+                                    except TypeError:
+                                        logger.debug('Type Error when attempting to use the given key, code will next attempt to convert the key to an int')
+                                    except KeyError:
+                                        msg = "Invalid channel address path for NTTable given. %s"
+                                        logger.exception(msg, self.nttable_data_location, exc_info=True)
+                                        raise KeyError("error in channel address")
+                                    
+                                    try:
+                                        new_value = new_value[int(value)] 
+                                    except ValueError:
+                                        msg = "Invalid channel address path for NTTable given. %s"
+                                        logger.exception(msg, self.nttable_data_location, exc_info=True)
+                                        raise ValueError("error in channel address")
+                            else:
+                                msg = "Invalid channel address path for NTTable given. %s"
+                                logger.exception(msg, self.nttable_data_location, exc_info=True)
+                                raise ValueError("error in channel address")
```

Comment:
Were these meant to be `raise ValueError(msg)` by chance?

Reply:
I think originally, but it doesn't like the old formatting style %s, but I could pass msg[:-3]

Reply:
Or maybe `msg = f"Invalid channel... {self.nttable_data_location}"`  and then reuse that in both spots? 🤷‍♂️